apps/processing/src/services/processing.service.ts

```diff
@@ -23,34 +31,63 @@ export class ProcessingService {
     private readonly logger = new Logger({ className: "ProcessingService" });
     private readonly kyselyDatabase: SharedDependencies["kyselyDatabase"];
 
-    private constructor(env: Environment, sharedDependencies: SharedDependencies) {
-        const { CHAINS: chains } = env;
-        const { core, registries, indexerClient, kyselyDatabase } = sharedDependencies;
+    private constructor(
+        orchestrators: Map<ChainId, Orchestrator>,
+        kyselyDatabase: SharedDependencies["kyselyDatabase"],
+    ) {
+        this.orchestrators = orchestrators;
         this.kyselyDatabase = kyselyDatabase;
+    }
+
+    static async initialize(env: Environment): Promise<ProcessingService> {
+        const sharedDependencies = await SharedDependenciesService.initialize(env);
+        const { CHAINS: chains } = env;
+        const { core, registriesRepositories, indexerClient, kyselyDatabase } = sharedDependencies;
+        const { eventRegistryRepository, strategyRegistryRepository } = registriesRepositories;
+        const orchestrators: Map<ChainId, Orchestrator> = new Map();
+
+        const strategyRegistry = await InMemoryCachedStrategyRegistry.initialize(
+            new Logger({ className: "InMemoryCachedStrategyRegistry" }),
+            new DatabaseStrategyRegistry(
+                new Logger({ className: "DatabaseStrategyRegistry" }),
+                strategyRegistryRepository,
+            ),
+        );
+        const eventsRegistry = new DatabaseEventRegistry(
+            new Logger({ className: "DatabaseEventRegistry" }),
+            eventRegistryRepository,
+        );
```

Comment:
is there a reason why you are doing this outside the `SharedDependenciesService` ?